### ISSUE TITLE
Many Sonarcloud alerts

### ISSUE THREAD

**Issue by 304NotModified** [NLog/NLog#4429]
The rules have been updated @ Sonarcloud  which results in many code smells and "bug" 

- master: https://sonarcloud.io/dashboard?id=nlog2
- dev: https://sonarcloud.io/dashboard?branch=dev&id=nlog2

![image](https://user-images.githubusercontent.com/5808377/118324855-e61aa500-b502-11eb-98c8-3d44b919678e.png)

**Comment by 304NotModified** [comment]
"bugs" and security hotspots handled

**Comment by snakefoot** [comment]
Yes it seems that these 2 are new:

![image](https://user-images.githubusercontent.com/11509660/118388553-8ec12580-b625-11eb-815e-ac5f7699e41b.png)

![image](https://user-images.githubusercontent.com/11509660/118388581-9ed90500-b625-11eb-94ee-d44cde16955f.png)

**Comment by snakefoot** [comment]
Before starting the work of fixing code-smells, then it would be nice to have completed the extraction of NLog-DatabaseTarget: #4351

**Comment by 304NotModified** [comment]
Ok, I only changes things I sonarcloud registration so far. It needs a rescan of the branches to see the differences. (disabled some new rules)

**Comment by 304NotModified** [comment]
no concrete plans for this now
